meetings/2025-05/may-29.md

```diff
@@ -0,0 +1,1490 @@
+
+LCA: Yeah, this is not a very exhaustive bit of data, but I just did a quick search of GitHub and Rust for the chunks and windows function in the standard library, and chunks is used 90,000 times in public repositories and windows is used 41,000 times in windows repositories, so they’re both in the same magnitude of order of use. I—yeah.
+
+CM: Not exactly a JavaScript application, but just in terms of general usefulness, this kind of thing comes up a lot in certain image processing applications, where you literally need to scan through imagery, a lot of image sources are serial data sources. Sufficiently useful that in my own history, I’ve been involved with projects which actually had to implement exactly this API in hardware. So I know it’s useful.
+
+DRR: I guess my input is that I just have a preference for 1 or possibly 2, but recognize that, you know, opting into the I want a smaller chunk behavior is not always possible if you don’t have a collection or an iterator that can actually get its length.
+
+MF: It sounds like we’re leaning in the direction that I thought that we were going to go, which is probably we should provide both number 1 and number 3 here as two separate methods. So given that, I do not have prepared pull requests for every combination of options that we had, so that means that I will go back and revise this proposal to have both of those. I will be happy to take, like, naming suggestions and I’ll be inspired by what exists out there in the libraries and languages that I reviewed.
+
+MF: If anybody would oppose such a direction, I would like to hear it now. It’s not a commitment to going that direction, of course, but I would like to hear now so I don’t waste my time doing that work.
+
+JHD: So I just threw an item on the queue. I don’t actually have a strong preference between separate methods or as a parameter. But, like, I can—every two minutes, I can go back and forth on which one I lean towards, so I was wondering about the committee’s thoughts on those two options. Obviously you could make pull requests for both options, MF, and we could talk about it at the next meeting. With you if we have a sense there’s a direction we should go in that way, we can skip the rigamarole.
+
+MF: I would prefer not to do double work again, and I have a preference personally for the separate methods over the parameter, just because it eliminates that “what was that parameter name or what was the possible parameter?” They’re two different methods and they’ll auto complete in the text editor.
+
+JHD: The argument is auto complete as well. Like, it pops up that, like, the description of what the parameters are and stuff.
+
+JHD: Well, maybe we can do some offline discussion versus the methods versus parameter,
+
+### Speaker's Summary of Key Points
+
+- The iterator chunking proposal has had several open design questions which are holding back further development
+- In particular, distinguishing between
+
```

Comment:
```suggestion
```

I'm not sure who wrote this, but I would prefer it not be attributed to me.

Reply:
* iterator chunking methods now close the receiver on argument validation failure, matching the recent change to all other iterator helpers
* presented some options for behaviour of `windows` method when the underlying iterator does not yield enough items to fill the first window